lucille-plugins/lucille-tika/src/main/java/com/kmwllc/lucille/tika/stage/TextExtractor.java

```diff
@@ -0,0 +1,158 @@
+import com.kmwllc.lucille.util.FileUtils;
+import com.typesafe.config.Config;
+import java.io.ByteArrayInputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Iterator;
+import java.util.List;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.apache.tika.config.TikaConfig;
+import org.apache.tika.exception.TikaException;
+import org.apache.tika.metadata.Metadata;
+import org.apache.tika.parser.AutoDetectParser;
+import org.apache.tika.parser.ParseContext;
+import org.apache.tika.parser.Parser;
+import org.apache.tika.sax.BodyContentHandler;
+import org.xml.sax.ContentHandler;
+import org.xml.sax.SAXException;
+
+/**
+ * This stage uses Apache Tika to perform text and metadata extraction
+ * <br>
+ * Config Parameters -
+ * <br>
+ * text_field (String, Optional) : name of destination field for parsed data to be placed
+ * file_path_field (String, Optional) : name of field from which file path can be extracted, if file_path_field
+ * and byte_array_field both not provided, stage will do nothing
+ * byte_array_field (String, Optional) : name of field from which byte array data can be extracted
+ * tika_config_path (String, Optional) : path to tika config, if not provided will default to empty AutoDetectParser
+ * metadata_prefix (String, Optional) : prefix to be appended to fields for metadata information extracted after parsing
+ * text_content_limit (Integer, Optional) : limits how large the content of the returned text can be
+ * metadata_whitelist (StringList, Optional) : list of metadata names that are to be included in document
+ * metadata_blacklist (StringList, Optional) : list of metadata names that are not to be included in document
+ *
+ * TODO
+ * - add test case for empty metadata_prefix, if metadata_prefix is empty,
+ *  do not include underscore i.e. "" -> content_type not _content_type
+ *
+ */
+public class TextExtractor extends Stage {
+
+  private static final Logger log = LogManager.getLogger(TextExtractor.class);
+  private String textField;
+  private String filePathField;
+  private String tikaConfigPath;
+  private String byteArrayField;
+  private String metadataPrefix;
+  private Integer textContentLimit;
+  private List<String> metadataWhitelist;
+  private List<String> metadataBlacklist;
+  private Parser parser;
+  private ParseContext parseCtx;
+
+  public TextExtractor(Config config) throws StageException {
+    super(config, new StageSpec().withOptionalProperties("text_field", "file_path_field", "byte_array_field", "tika_config_path",
+        "metadata_prefix", "metadata_whitelist", "metadata_blacklist", "text_content_limit"));
+    textField = config.hasPath("text_field") ? config.getString("text_field") : "text";
+    filePathField = config.hasPath("file_path_field") ? config.getString("file_path_field") : "filepath";
```

Comment:
should the default value be "file_path" with the underscore considering the default byte array field is "byte_array"?

Reply:
I agree, makes more sense with the underscore, change added in new commit